django_stomp/execution.py

```diff
@@ -149,3 +150,15 @@ def _create_queue(queue_name: str, durable_topic_subscription: bool = False, rou
 def _create_dlq_queue(destination_name: str):
     dlq_destination_name = create_dlq_destination_from_another_destination(destination_name)
     _create_queue(dlq_destination_name)
+
+
+def _get_or_create_correlation_id(payload: Payload) -> str:
+    if "correlation-id" in payload.headers:
+        return payload.headers["correlation-id"]
+
+    if not is_correlation_id_required:
+        correlation_id = uuid.uuid4()
+        logger.info(f"STOMP_CORRELATION_ID_REQUIRED was set to false. New correlation-id was generated {correlation_id}")
+        return correlation_id
+
+    raise Exception(f"correlation-id header is required! Headers: {payload.headers}")
```

Comment:
I'm not a big fan of bare Exception, here I suggest to use something like `CorrelationIdNotProvidedError` and feed it with `headers`, it's self explanatory.